### PR TITLE
Heretic/Hexen mouselook

### PR DESCRIPTION
**Pull request by mikeday0**
Even though it's not finished, I would appreciate any early comments on what I've done so far on mouselook. It seems too simple and I'm afraid I may have missed something. Also, Heretic now has two Crispness pages! (Sorry, no birthday cake.)

### CODE REVIEW

File: src/heretic/g_game.c [hunk 1]
```diff
@@ -690,8 +690,15 @@ void G_BuildTiccmd(ticcmd_t *cmd, int maketic)
         testcontrols_mousespeed = 0;
     }
 
-    if (!novert)
+    if (crispy->mouselook)
+    {
+        cmd->lookdir = mouse_y_invert ? -mousey : mousey;
+        cmd->lookdir /= 8;
```
Comment (fabiangreffrath) [inline src/heretic/g_game.c:696]: Could you keep/introduce the MLOOKDIR macro?

File: src/heretic/mn_menu.c [hunk 2]
```diff
@@ -661,7 +681,8 @@ void MN_Drawer(void)
         {
             if (item->type != ITT_EMPTY && item->text)
             {
-                if (CurrentMenu == &CrispnessMenu)
+                if (CurrentMenu == &Crispness1Menu ||
```
Comment (fabiangreffrath) [inline src/heretic/mn_menu.c:684]: You could either macro'ize it. Or, in Doom, I have introduced a pointer that always points to the current Crispness menu page.

Comment (mikeday0) [reply src/heretic/mn_menu.c]: I redid the Crispness menu code and this check is much simpler now.

**Review by fabiangreffrath — APPROVED** [review]

**Comment by JNechaevsky** [comment]
@mikeday0 👋, do you have plans of changing Crispness page №1 in Hexen? I'm planning to add brightmaps to Hexen, but probably, it will be better I'll wait for you to finish so we will not have a commit race?

**Comment by fabiangreffrath** [comment]
> @mikeday0 👋, do you have plans of changing Crispness page №1 in Hexen? I'm planning to add brightmaps to Hexen, but probably, it will be better I'll wait for you to finish so we will not have a commit race?

I think it should be no problem if you submit the implementation without the menu entry and just add it later when Mike has finished his work. That'd be one `git merge master` in between.

**Comment by mikeday0** [comment]
> @mikeday0 👋, do you have plans of changing Crispness page №1 in Hexen? I'm planning to add brightmaps to Hexen, but probably, it will be better I'll wait for you to finish so we will not have a commit race?

Yes, I plan to add the Tactical section with Enable Permanent Mouselook option. Hoping to finish everything tonight. Don't let me hold you up though. It should be no problem to `git merge` (or `git rebase` if Fabian doesn't mind rewriting history on a branch) to get my changes in your branch or vice-versa.

**Comment by JNechaevsky** [comment]
Understood, thanks! Hexen brightmaps are very straight-forward, but I still need some extra time to double check is it possible to apply all of them w/o sprites correction.

**Comment by mikeday0** [comment]
Okay, I think it's done! Just let me know if I need to merge Julia's brightmap commits before merging this PR into master.

**Review by fabiangreffrath — APPROVED** [review]

**Comment by JNechaevsky** [comment]
I'm late, but better now than never! Exactly two, but important recommendations - reset looking direction to straight when:
* Mouse look is turned off.
* Player is dead (just like in Crispy Doom).

**Comment by JNechaevsky** [comment]
I'm late, but better now than never! Exactly two, but important recommendations - reset looking direction to straight when:
* Mouse look is turned off.
* Player is dead (just like in Crispy Doom).

**Comment by mikeday0** [comment]
> I'm late, but better now than never! Exactly two, but important recommendations - reset looking direction to straight when:
> 
> * Mouse look is turned off.
> * Player is dead (just like in Crispy Doom).

Thanks Julia! I will definitely implement these suggestions when I am back home next week. I also need to add mouse-button enable for mouselook, mouse y inversion and lookspring to bring HH to the same level as Doom.

**Pull request merged**